Core/Core/Dashboard/K5/View/Grades/K5GradesView.swift

```diff
@@ -19,7 +19,87 @@
 import SwiftUI
 
 struct K5GradesView: View {
+
+    @ObservedObject private var viewModel: K5GradesViewModel
+    @State var gradeSelectorOpen = false
+
+    init(viewModel: K5GradesViewModel) {
+        self.viewModel = viewModel
+    }
+
     var body: some View {
-        Text("Grades Details", bundle: .core)
+        VStack(spacing: 0) {
+            VStack(alignment: .leading) {
+                Button(action: {
+                    withAnimation {
+                        gradeSelectorOpen.toggle()
+                    }
+                }, label: {
+                    VStack(alignment: .leading, spacing: 0) {
+                        Text("Select", bundle: .core)
+                            .font(.regular13)
+                            .background(Color.white)
+                            .foregroundColor(.ash)
+                            .padding(.top, 15)
+                        HStack(spacing: 7) {
+                            Text(viewModel.currentGradingPeriod.title ?? "")
+                                .font(.bold24)
+                                .foregroundColor(.licorice)
+                            Image.arrowOpenDownLine
+                                .resizable()
+                                .frame(width: 12, height: 12)
+                                .foregroundColor(.licorice)
+                                .rotationEffect(.degrees(gradeSelectorOpen ? -180 : 0))
+                                .animation(.easeOut)
+                            Spacer()
+                        }
+                        .padding(.bottom, 13)
+                        Divider()
+                    }
+                })
+                .zIndex(1)
+                .background(Color.white)
+                .padding(.top, 0)
+
+                if gradeSelectorOpen {
+                    VStack(alignment: .leading) {
+                        ForEach(viewModel.gradingPeriods, id: \.self) {
+                            let gradingPeriod = $0
```

Comment:
We could write `ForEach(viewModel.gradingPeriods, id: \.self) { gradingPeriod in` instead.